vm/src/vm.rs

```diff
@@ -397,7 +397,11 @@ impl VirtualMachine {
 
     #[cfg(feature = "rustpython-compiler")]
     pub fn new_syntax_error(&self, error: &CompileError) -> PyObjectRef {
-        let syntax_error_type = self.ctx.exceptions.syntax_error.clone();
+        let syntax_error_type = if error.is_tab_error() {
+            self.ctx.exceptions.tab_error.clone()
+        } else {
+            self.ctx.exceptions.syntax_error.clone()
+        };
```

Comment:
Disclaimer that you probably shouldn't take my advice unless @coolreader18 or one of the other long-term devs says it's good, because I'm new to the project, and I don't know if there's a better way to do it than my suggestions :)

But anyway... I'm not sure this is the right place to put this fix.  ```new_syntax_errror``` should probably not sometimes  raise a ```TabError```.

 I think, instead, what you probably want to do is to add a new ```CompileErrorType``` here: https://github.com/RustPython/RustPython/blob/master/compiler/src/error.rs#L22-L61

And add new methods here for ```new_tab_error``` and ```new_indentation_error```.

And then perform a match here to raise the correct error type: https://github.com/RustPython/RustPython/blob/master/vm/src/eval.rs#L12

And then you could get rid of this block and the ```impl CompileError``` block up above.

Reply:
I personally think this is fine, technically `new_syntax_error` still always returns a `SyntaxError`, given that `IndentationError` and `TabError` are subclasses of it, and `new_syntax_error` is the main point of conversion between Rust struct/enum representation of compilation errors and Python exceptions, so if the conversion is here it will always catch it.

Reply:
@coolreader18 answered it better than I could. I don't see a need to change my code.

Reply:
Fair!  I'm still learning the codebase myself, hence the disclaimer :)